File: cracking-the-code.py
from math import sqrt
a=raw_input().split(' ')
a=map(int,a)
b=raw_input().split(' ')
b=map(int,b)
m=(a[1]-a[0])%2
l=4*int((a[1]-a[0])/2)+m-a[2]
print ("%.4f" % round(pow(2,l)*pow(sqrt(2),m)*(b[0]+b[1]*pow(sqrt(3),m)),4))

File: cracking-the-code.cpp
#include <iostream>
#include <math.h>
#include <stdio.h>

using namespace std;

int main(){
	long i,k,s,a,b,m,l;
	cin >> i >> k >> s >> a >> b;
	l=int(abs(k-i)/2);
	m=abs(k-i)%2;
	long double z;
	if(k>=i){
		z=pow(2,4*l+m-s)*pow(sqrt(2),m)*(a+b*pow(sqrt(3),m));
	}
	else{
		z=pow(2,-4*l-2*m-s)*pow(sqrt(2),-m)*(a+b*pow(sqrt(3),m));
	}
	printf("%.4Lf\n",z);
}